### PR TITLE
[52] [Backend] As a logged in user, I can fetch a survey detail

### LINKED ISSUE #52: [Backend] As a logged in user, I can fetch a survey detail
## Why

- Fetch survey question list in order to show on survey question screen.
 
## Acceptance Criteria

- Call API endpoint `/api/v1/surveys/{{surveyId}}`
- After successful request decode the response to object for use later.

### PR DESCRIPTION
**Pull request by chiiida**
https://github.com/llleyelll/ic-surveys-ios/issues/52

## What happened 👀

- Implement fetch survey detail
 
## Insight 📝

- Update the `SurveyResponse` model since `SurveyDetailResponse` has a similar model, only attribute of question and answer that different.
 
## Proof Of Work 📹

Screen Shot 2021-07-05 at 9 24 54 AM

### CODE REVIEW

File: SurveyApp/Models/Survey/SurveyResponse.swift [hunk 1]
```diff
@@ -7,7 +7,9 @@
 
 import Foundation
 
-struct SurveyAttributes: Codable {
+protocol SurveyAttributeType {}
```
Comment (blyscuit) [inline SurveyApp/Models/Survey/SurveyResponse.swift]: Since there's no common variables in the classes, I think we can jut map to Any?

**Review by blyscuit — APPROVED** [review]

File: SurveyApp/Models/Survey/SurveyResponse.swift (hunk 1 above)
Comment (chiiida) [reply SurveyApp/Models/Survey/SurveyResponse.swift]: I updated the code. Thank you for a better approach 🙇‍♀️

File: SurveyApp/Modules/Home/HomeInteractor.swift [hunk 2]
```diff
@@ -49,15 +49,15 @@ extension HomeInteractor: HomeInteractorInput {
             switch result {
             case .success(let surveyResponse):
                 let surveys: [Survey] = surveyResponse.data.map {
+                    let attributes = $0.attributes as? SurveyAttributes
```
Comment (markgravity) [inline SurveyApp/Modules/Home/HomeInteractor.swift:52]: We should remove any element whose `attributes` is `nil`, right? A survey without `title`, `description`, and `coverImageUrl` seem not a valid survey to display.

You can use [compactMap](https://developer.apple.com/documentation/swift/sequence/2950916-compactmap) here

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**